#!/usr/bin/env python

PACKAGE = "pedsim_simulator"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('update_rate', double_t, 0,
        'Simulation frequency (Hz)', 25.0, 1.0, 50.0)
gen.add('simulation_factor', double_t, 0, 'Simulation factor', 1.0, 0.1, 10.0)

gen.add('force_obstacle', double_t, 0,
        'Obstacle force weight', 10.0, 0.0, 50.0)
gen.add('sigma_obstacle', double_t, 0,
        'Sigma factor (obstacles)', 0.2, 0.1, 1.0)
gen.add('force_social', double_t, 0, 'Social force weight', 5.1, 0.0, 10.0)
gen.add('force_group_gaze', double_t, 0,
        'Group gaze force weight', 3.0, 0.0, 10.0)
gen.add('force_group_coherence', double_t, 0,
        'Group coherence force weight', 2.0, 0.0, 10.0)
gen.add('force_group_repulsion', double_t, 0,
        'Group repulsion force weight', 1.0, 0.0, 10.0)
gen.add('force_random', double_t, 0, 'Random force weight', 0.1, 0.0, 1.0)
gen.add('force_wall', double_t, 0, 'Wall force weight', 2.0, 0.0, 10.0)

gen.add('robot_mode', int_t, 0,
        'Robot operation mode (0 = controlled, 1 = teleop, 2 = social drive)',
        0, 0, 2)
gen.add('robot_wait_time', int_t, 0,
        'Robot waiting time before moving', 15, 1, 100)
gen.add('max_robot_speed', double_t, 0,
        'Robot maximum speed (m/s)', 1.8, 0.5, 4.5)

gen.add('enable_groups', bool_t, 0, 'Enable forming of groups ', True)

gen.add('cell_width', double_t, 0, 'Grid cell size', 1.0, 0.1, 10.0)
gen.add('cell_height', double_t, 0, 'Grid cell size', 1.0, 0.1, 10.0)

gen.add('group_size_lambda', double_t, 0,
        'Group size distribution lambda parameter', 1.1, 0.0, 3.0)
gen.add('wait_time_beta', double_t, 0,
        'Queue waiting distribution beta parameter', 0.2, 0.0, 1.0)

gen.add('paused', bool_t, 0, 'Pause/unpause simulation', False)


exit(gen.generate(PACKAGE, PACKAGE, "PedsimSimulator"))
